### ISSUE TITLE
Admin user trying to get the personal drive information of other users returns 404

### ISSUE THREAD

**Issue by PrajwolAmatya** [owncloud/ocis#6017]
## Describe the bug

When admin user tries to get the personal drive information of other users using the endpoint `/graph/v1.0/users/{userid}/drive`, then the returned status code is `404`.

## Steps to reproduce

1. Create users `Alice` and `Brian`
2. Assign `Alice` the role `Admin`
3. `Alice` tries to get the drive information of `Brian` using the endpoint `/graph/v1.0/users/{userid}/drive`

## Actual Behavior
The returned status code is `404`.
```
> GET /graph/v1.0/users/4c510ada-c86b-4815-8820-42cdf82c3d51/drive HTTP/1.1
> Host: localhost:9200
> Authorization: Basic YWRtaW46YWRtaW4=
> User-Agent: curl/7.81.0
> Accept: */*
> 
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* TLSv1.3 (IN), TLS handshake, Newsession Ticket (4):
* TLSv1.2 (IN), TLS header, Supplemental data (23):
* Mark bundle as not supporting multiuse
< HTTP/1.1 404 Not Found
< Content-Length: 177
< Content-Security-Policy: frame-ancestors 'none'
< Content-Type: application/json; charset=utf-8
< Date: Wed, 12 Apr 2023 08:30:50 GMT
< Vary: Origin
< X-Content-Type-Options: nosniff
< X-Frame-Options: DENY
< X-Graph-Version: 3.0.0-alpha.2+0e2d20fe6
```

## Expected Behavior
The returned status code should be `200`. And the response should contain the personal drive information of user `Brian`.

**Comment by micbar** [comment]
@PrajwolAmatya Did Brian log in at least once?

Brian will have no personal drive if he hasn't successfully logged in yet.

**Comment by PrajwolAmatya** [comment]
> @PrajwolAmatya Did Brian log in at least once?
> 
> Brian will have no personal drive if he hasn't successfully logged in yet.

I tried with Brian logging in, the returned response is `404`.

**Comment by PrajwolAmatya** [comment]
When, user `Brian` tries to get his own personal drive information then the status code is `200` and the response contains his drive information.

**Comment by micbar** [comment]
I found the problem. Committed a fix to #6018

**Comment by PrajwolAmatya** [comment]
> I found the problem. Committed a fix to #6018

@micbar with the fix, now in case of `Space Admin` trying to get personal drive information of other users, it is returning `200` status code.

**Comment by saw-jan** [comment]
From https://github.com/owncloud/ocis/pull/6018#issuecomment-1516310920
> IMO this should behave the same like for users with the "Admin" role.
